src/mapgen/mapgen.cpp

```diff
@@ -553,19 +552,18 @@ void Mapgen::spreadLight(v3s16 nmin, v3s16 nmax)
 
 				u8 light = n.param1;
 				if (light) {
-					lightSpread(a, queue, v3s16(x,     y,     z + 1), light);
-					lightSpread(a, queue, v3s16(x,     y + 1, z    ), light);
-					lightSpread(a, queue, v3s16(x + 1, y,     z    ), light);
-					lightSpread(a, queue, v3s16(x,     y,     z - 1), light);
-					lightSpread(a, queue, v3s16(x,     y - 1, z    ), light);
-					lightSpread(a, queue, v3s16(x - 1, y,     z    ), light);
+					const v3s16 p(x, y, z);
+					// spread to all 6 neighbor nodes
+					for (u8 i = 0; i < 6; i++) {
+						lightSpread(a, queue, p + g_6dirs[i], light);
+					}
```

Comment:
```c++
for (const auto &dir : g_6dirs)
    lightSpread(a, queue, p + dir, light);
```

Reply:
Huh, TIL that ranged for loops are possible for normal arrays.
Done.